### ISSUE TITLE
browser launchOptions in config don't seem to flow through to launcher

### ISSUE THREAD

**Issue by davemooreuws** [modernweb-dev/web#666]
I am trying to run test-runner with puppeteer on a linux box.

My dependencies:
 "@web/dev-server-esbuild": "^0.2.4",
 "@web/test-runner": "^0.7.38",
 "@web/test-runner-puppeteer": "^0.6.4",

This is my config:

```js
const { esbuildPlugin } = require("@web/dev-server-esbuild");
const { puppeteerLauncher } = require("@web/test-runner-puppeteer");

module.exports = {
  browsers: [
    puppeteerLauncher({ launchOptions: { args: ["--no-sandbox", "--disable-setuid-sandbox"] } }),
  ],
  plugins: [esbuildPlugin({ ts: true })],
  nodeResolve: true,
  sessionStartTimeout: 30000,
  concurrency: 5,
  coverageConfig: {
    report: true,
    reportDir: "coverage",
    threshold: {
      statements: 90,
      branches: 65,
      functions: 80,
      lines: 90,
    },
  },
  testFramework: {
    config: {
      timeout: "3000",
    },
  },
};
```

I get the following error on the box:
`Error: Failed to launch the browser process! 
      [1001/030354.274765:FATAL:zygote_host_impl_linux.cc(117)] No usable sandbox! Update your kernel or see https://chromium.googlesource.com/chromium/src/+/master/docs/linux/suid_sandbox_development.md for more information on developing with the SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.`

I had a look through the test-runner code. I can't see where these args are being passed in to the final config.
Example: https://github.com/modernweb-dev/web/blob/37d660de1be733d7010f0d92e03d2ba22925f759/packages/test-runner/src/loadLauncher.ts#L32 seems to just pass product back.

**Comment by LarsDenBakker** [comment]
I think you are using the `--puppeteer` flag? This flag is meant as a shorthand to run the browser with defaults, but I see that it's always overwriting the user defined browsers option. 

Can you try without the flag?

I made sure we throw an error in this case: https://github.com/modernweb-dev/web/pull/673

**Comment by davemooreuws** [comment]
Thank you for your quick reply.

Yes I was using the `--puppeteer` flag.
I have tried without the flag and it works.

As well as your checks in the code, I think it might be a good idea to update the docs here: 
https://modern-web.dev/docs/test-runner/browser-launchers/overview/#configuring-browser-launchers
It isn't 100% obvious to not use the flag when customizing the browsers as I think there is a typo "confusing" to "customizing"?
Maybe put a note: don't use the default flags if using this feature.

**Comment by LarsDenBakker** [comment]
Yep you're definitely right, we can make this clearer. And it looks like a typo indeed 👍
